src/js/polyfills/chrome.js

```diff
@@ -0,0 +1,268 @@
+        //   'data': {},
+        //   'randomMessageID': 'random string',
+        //   'sender': {
+        //     'frameId': 'frame id',
+        //     'tabId': {
+        //       'id':'tab id'
+        //     }
+        //   }
+        // }
+
+        // if message is from the content script
+        // that means that this script is a background script
+        if (data.data.event == 'connectPort') {
+          // a client wants to establish a connection
+          // run all listeners waiting for a connection
+          // and pass it a fake port
+          const portID = data.data.portID;
+          const port = {
+            sender: data.sender,
+            postMessage: data => {
+              sendToForeground({
+                event: 'postMessage',
+                portID,
+                data: data
+              }, data.randomMessageID);
+            },
+            onMessage: {
+              addListener: callback => {
+                const portCallbacks = polyfillStorage.portOnMessageCallbacks[portID];
+                polyfillStorage.portOnMessageCallbacks[portID] = portCallbacks || [];
+                polyfillStorage.portOnMessageCallbacks[portID].push(callback);
+              }
+            },
+            onDisconnect: {
+              // eslint-disable-next-line no-unused-vars
+              addListener: callback => { } // can't really detect disconnect so just ignore
+            }
+          }; // construct a fake port
+          // pass the fake port to all listeners that were waiting for a port connection
+          polyfillStorage.onConnectCallbacks.forEach(callback => {
+            try {
+              callback(port);
+            } catch (error) {
+              console.debug('Error in callback', error);
+            }
+          });
+          return;
+        } else if (data.data.event == 'postMessage') {
+          // received a postMessage sent through a port.
+          (polyfillStorage.portOnMessageCallbacks[data.data.portID] || []).forEach(
+            callback => {
+              try {
+                callback(data.data.data);
+              } catch (error) {
+                console.debug('Error while running callback in port postMessage receiver', error);
+              }
+            }
+          ); // pass the data to all listeners of this port
+          return;
+        }
+      }
+      // if it's none of the above, that means it's a standalone message
+      // (which i don't think happens in the content script but i'll support it anyway)
+      if (data.type == 'sendToForeground' || data.type == 'sendToBackground') {
+        polyfillStorage.onMessageCallbacks.forEach(callback => {
+          try {
+            callback(data.data, data.sender, d => sendToForeground(d, data.randomMessageID));
+          } catch (error) {
+            console.debug('Error while running callback in standalone receiver', error);
+          }
+        });
+      }
+    } catch (e) {
+      console.debug(e);
```

Comment:
Add a descriptive error messages like `Errror responding to postmessage in chrome polyfill`. You should also use ``console.error``.